### PR TITLE
401 example 4 is not working properly

### PR DESCRIPTION
**Pull request by lovfall**
Implemented working example based on C example 1. 
This PR depends on https://github.com/SINTEF/dlite/pull/414

Clean up and documentation is still missing, this PR should not be merged before that is done

### CODE REVIEW

File: examples/ex4/CMakeLists.txt [hunk 1]
```diff
@@ -2,36 +2,51 @@
...
   ${DLITE_ROOT}/cmake
   ${DLITE_ROOT}/cmake/Modules
   ${DLITE_ROOT}/share/cmake
   ${DLITE_ROOT}/share/cmake/Modules
   )
-include(Finddlite)
-find_package(dlite)
 
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${DLITE_ROOT}/share/dlite/cmake)
```
Comment (jesper-friis) [inline examples/ex4/CMakeLists.txt:23]: This can be included in the `list(APPEND ...)` above

**Review by jesper-friis — APPROVED** [review]

**Review by jesper-friis — CHANGES_REQUESTED** [review]
The github CI should be fixed

File: examples/ex4/CMakeLists.txt (hunk 1 above)
Comment (lovfall) [reply examples/ex4/CMakeLists.txt:25]: Have no clue why, but that does not work

Comment (jesper-friis) [reply examples/ex4/CMakeLists.txt:25]: Ok, then we keep it

**Review by jesper-friis — APPROVED** [review]

**Pull request merged**